### PR TITLE
support code cli

### PR DESCRIPTION
**Pull request by akosyakov**
- [x] /werft https

#### What it does

- enables IDE images to contribute env variables via special scheme:
```dockerfile
# to prepend a value to KEY variable value
ENV GITPOD_ENV_PREPEND_KEY "value" 
# to append a value to KEY variable value
ENV GITPOD_ENV_APPEND_KEY "value" 
# to set a value for KEY variable
ENV GITPOD_ENV_SET_KEY "value" 
```
- implements code CLI: https://github.com/gitpod-io/vscode/compare/20ded0d...785b788
- configure Gitpod Code image using env variables.

#### How to test

- With disabled feature preview start a workspace and check that `gp`, `code` and `open` are available as well as `EDITOR`, `VISUAL` and `GIT_EDITOR` env variables are configured to use them.
  - Check both in task terminals started by the supervisor and regular terminals started by Theia. It should be true for tests below as well.
- Enable feature preview and start a workspace with as well. Do checks again.
- Now switch to Code as a default IDE and start a new workspace. Check taht `code` command does not delegate to `gp` anymore. Check that `code`, `open` and `gitpod-code` do the same. Check that env variables are configured properly.
  - Don't forget that different options of code command actually work.

### CODE REVIEW

**Comment by svenefftinge** [comment]
foxes #2625

File: components/registry-facade/pkg/registry/layersource.go [hunk 1]
```diff
@@ -489,3 +559,85 @@ func (src *ContentLayerSource) GetBlob(ctx context.Context, spec *api.ImageSpec,
 	err = errdefs.ErrNotFound
 	return
 }
+
+// ParsedEnvs is parsed image envs configuration
+type ParsedEnvs struct {
```
Comment (csweichel) [inline components/registry-facade/pkg/registry/layersource.go:564]: nit: it's a bit odd that `ParsedEnv` has no exposed fields or functions. I'd make `set`, `append`, and `prepent` public s.t. others can implement their own `EnvModifier`

File: components/registry-facade/pkg/registry/layersource.go [hunk 2]
```diff
@@ -245,7 +260,35 @@ func NewStaticSourceFromImage(ctx context.Context, resolver remotes.Resolver, re
 		res = append(res, l)
 	}
 
-	return res, nil
+	var envs []EnvModifier
+	parsedEnvs := parseEnvs(cfg.Config.Env)
+	for _, name := range parsedEnvs.keys {
+		value := parsedEnvs.values[name]
+		if strings.HasPrefix(name, "GITPOD_APPEND_ENV_") {
```
Comment (csweichel) [inline components/registry-facade/pkg/registry/layersource.go:267]: nit: I'd probably name the prefixes `GITPOD_ENV_SET_`, `GITPOD_ENV_APPEND_` and `GITPOD_ENV_PREPEND` and put them in a const section:
```Go
const (
    envPrefixSet = "GITPOD_ENV_SET_"
    envPrefixAppend = "GITPOD_ENV_APPEND_"
    envPrefixPrepend = "GITPOD_ENV_PREPEND_"
)
```

**Review by csweichel — APPROVED** [review]
Love this PR, and it works like a charm.

I have comment regarding `GITPOD_SET_ENV` env var prefix. For consistency's sake, I'd vote for `GITPOD_ENV_SET`, `GITPOD_ENV_APPEND`, `GITPOD_ENV_PREPEND`. Considering that we're making this "API" we should decide this prior to landing it on master.

**Pull request merged**